packages/data/src/redux-store/metadata/selectors.js

```diff
@@ -54,6 +71,36 @@ export function hasFinishedResolution( state, selectorName, args = [] ) {
 	return getIsResolving( state, selectorName, args ) === false;
 }
 
+/**
+ * Returns true if resolution has failed for a given selector
+ * name, and arguments set.
+ *
+ * @param {State}     state        Data state.
+ * @param {string}    selectorName Selector name.
+ * @param {unknown[]} [args]       Arguments passed to selector.
+ *
+ * @return {boolean} Has resolution failed
+ */
+export function hasResolutionFailed( state, selectorName, args = [] ) {
```

Comment:
Does it make sense to default `args` to `[]`? The existing `getIsResolving` selector doesn't do that. That's at least inconsistent: some selectors do default, other don't.

Reply:
I agree it's inconsistent. It follows the other consumers of `getIsResolving` like `hasFinishedResolution`, `isResolving`, and `hasStartedResolution`. That empty list is unfortunately required to retrieve the correct map entry, otherwise the consumers of this selector would have to specify that when needed.

That's beyond the scope of this PR, but let's consider how we could make it consistent.

One way would be to store the argument-less resolution state under the `undefined` key. I'd worry this it would break some consumers of this API out in the wild, though. E.g. it would break any code that calls `hasStartedResolution( selectorName, args )` where args has varying length of 0 or more.

The other alternative is to give all the selectors the default argument value of `[]`. It doesn't seem to make sense to explicitly pass undefined as that third parameter, so I don't think it would break anything at all.

Reply:
I (independently of this PR) discovered there are even some bugs in how the `args` array is handled and how the map entries are retrieved. I'm fixing that thoroughly in #38945.

Reply:
That's a great one! I just rebased this PR on top of trunk.